discovery/eureka/eureka.go

```diff
@@ -0,0 +1,221 @@
+	appInstancePortLabel                              = metaAppInstanceLabelPrefix + "port"
+	appInstancePortEnabledLabel                       = metaAppInstanceLabelPrefix + "port_enabled"
+	appInstanceSecurePortLabel                        = metaAppInstanceLabelPrefix + "secure_port"
+	appInstanceSecurePortEnabledLabel                 = metaAppInstanceLabelPrefix + "secure_port_enabled"
+	appInstanceDataCenterInfoNameLabel                = metaAppInstanceLabelPrefix + "datacenterinfo_name"
+	appInstanceDataCenterInfoMetadataPrefix           = metaAppInstanceLabelPrefix + "datacenterinfo_metadata_"
+	appInstanceDataCenterInfoMetadataAmiLaunchIndex   = appInstanceDataCenterInfoMetadataPrefix + "ami_launch_index"
+	appInstanceDataCenterInfoMetadataLocalHostname    = appInstanceDataCenterInfoMetadataPrefix + "local_hostname"
+	appInstanceDataCenterInfoMetadataAvailabilityZone = appInstanceDataCenterInfoMetadataPrefix + "availability_zone"
+	appInstanceDataCenterInfoMetadataInstanceID       = appInstanceDataCenterInfoMetadataPrefix + "instance_id"
+	appInstanceDataCenterInfoMetadataPublicIpv4       = appInstanceDataCenterInfoMetadataPrefix + "public_ipv4"
+	appInstanceDataCenterInfoMetadataPublicHostname   = appInstanceDataCenterInfoMetadataPrefix + "public_hostname"
+	appInstanceDataCenterInfoMetadataAmiManifestPath  = appInstanceDataCenterInfoMetadataPrefix + "ami_manifest_path"
+	appInstanceDataCenterInfoMetadataLocalIpv4        = appInstanceDataCenterInfoMetadataPrefix + "local_ipv4"
+	appInstanceDataCenterInfoMetadataHostname         = appInstanceDataCenterInfoMetadataPrefix + "hostname"
+	appInstanceDataCenterInfoMetadataAmiID            = appInstanceDataCenterInfoMetadataPrefix + "ami_id"
+	appInstanceDataCenterInfoMetadataInstanceType     = appInstanceDataCenterInfoMetadataPrefix + "instance_type"
+	appInstanceCountryIDLabel                         = metaAppInstanceLabelPrefix + "country_id"
+	appInstanceIDLabel                                = metaAppInstanceLabelPrefix + "id"
+	appInstanceMetadataPrefix                         = metaAppInstanceLabelPrefix + "metadata_"
+)
+
+// DefaultSDConfig is the default Eureka SD configuration.
+var DefaultSDConfig = SDConfig{
+	RefreshInterval: model.Duration(30 * time.Second),
+}
+
+// SDConfig is the configuration for applications running on Eureka.
+type SDConfig struct {
+	Server           string                       `yaml:"server,omitempty"`
+	RefreshInterval  model.Duration               `yaml:"refresh_interval,omitempty"`
+	HTTPClientConfig config_util.HTTPClientConfig `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultSDConfig
+	type plain SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	if len(c.Server) == 0 {
+		return errors.New("eureka_sd: empty or null eureka server")
+	}
+	_, err = url.Parse(c.Server)
+	if err != nil {
```

Comment:
That is not enough. Almost everything parses as a url. I'd like to explicitely have a check for the empty scheme.

Reply:
I'll check if the scheme is empty.